apps/web/scripts/update-web-ui-imports.ts

```diff
@@ -0,0 +1,361 @@
+                const newPath = IMPORT_MAP[importedName]
+
+                if (newPath) {
+                  mappedImports.push({ name: importedName, path: newPath })
+                } else {
+                  unmappedImports.push(specifier)
+                  console.warn(
+                    `Warning: No mapping found for ${importedName} in ${file}`,
+                  )
+                }
+              } else if (n.ImportDefaultSpecifier.check(specifier)) {
+                unmappedImports.push(specifier)
+              } else if (n.ImportNamespaceSpecifier.check(specifier)) {
+                unmappedImports.push(specifier)
+              }
+            })
+
+            // Group mapped imports by path
+            const importsByPath = mappedImports.reduce(
+              (acc, { name, path }) => {
+                if (!acc[path]) {
+                  acc[path] = []
+                }
+                acc[path].push(name)
+                return acc
+              },
+              {} as Record<string, string[]>,
+            )
+
+            // Create new import declarations for each path
+            const newImports = Object.entries(importsByPath).map(
+              ([path, names]) => {
+                return recast.types.builders.importDeclaration(
+                  names.map((name) =>
+                    recast.types.builders.importSpecifier(
+                      recast.types.builders.identifier(name),
+                      recast.types.builders.identifier(name),
+                    ),
+                  ),
+                  recast.types.builders.literal(
+                    `@latitude-data/web-ui/${path}`,
+                  ),
+                )
+              },
+            )
+
+            // If we have unmapped imports, create a new import for them
+            if (unmappedImports.length > 0) {
+              newImports.push(
+                recast.types.builders.importDeclaration(
+                  unmappedImports,
+                  recast.types.builders.literal('@latitude-data/web-ui'),
+                ),
+              )
+            }
+
+            // Replace the original import with all new imports
+            if (newImports.length > 0) {
+              path.replace(...newImports)
+              hasChanges = true
+            }
+          }
+
+          // Continue traversing the AST
+          this.traverse(path)
+        },
+      })
+
+      // If changes were made, write the updated code back to the file
+      if (hasChanges) {
+        const updatedCode = recast.print(ast).code
+        writeFileSync(filePath, updatedCode)
+        console.log(`Updated imports in ${file}`)
+      }
+    } catch (error) {
+      console.error(`Error processing file ${file}:`, error)
+      continue // Skip this file and continue with the next one
+    }
+  }
+}
+
+updateImports().catch((error) => {
+  console.error('Error updating imports:', error)
+  process.exit(1)
+})
```

Comment:
these are some recast scripts i've used to migrate the imports not sure if it's valuable to include them in the repo for future reference

Reply:
I wouldn't